storage/core/src/info.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Low-level runtime database layout description
+
+/// Database index type, just a thin wrapper over usize
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
+pub struct DbIndex(usize);
+
+impl DbIndex {
+    /// New index
+    pub const fn new(idx: usize) -> Self {
+        DbIndex(idx)
+    }
+
+    /// Get the index as usize
+    pub const fn get(&self) -> usize {
+        self.0
+    }
+}
+
+/// Description of one key-value store in a database
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub struct MapDesc {
+    /// Key-value map name
+    pub name: &'static str,
+    /// Value size hint
+    pub size_hint: core::ops::Range<usize>,
+    // TODO not implemented, is it even needed?
+    // /// Map multiplicity (i.e. is this a key-value or a key-multi-value map)
+    // pub multiplicity: Multiplicity,
```

Comment:
I can't really say whether this is needed right now. I needed it in other projects for tokenization, but we may have found a way around it in Mintlayer, still not sure. Though it's reasonable to have this. Does it make things much more complicated or something?

Reply:
To be honest I'd like to see if we can get away without multi-valued maps as it would make a number of things simpler. The backend code would be slightly more complicated with multi-valued maps but that I expect not to be a massive issue. In the high-level interface, I'd like to see a type-level distinction between single-valued and multi-valued maps and this would create a fair amount of ceremony around schema definitions.

If the need for this arises, I we could introduce it and think about how to do it nicely, but for the time of being I'd be in favour of getting rid of it. At the moment, there's some amount of dead code in service of feature that's not fully implemented.

Reply:
OK. Let's get rid of it for now. I can't promise we won't need it in the future though.